packages/perspective/src/js/perspective.js

```diff
@@ -1344,7 +1328,18 @@ export default function(Module) {
                 return `${match.substr(0, match.indexOf(full))}'${value}')`;
             });
 
-            validated_expressions.push([expression_alias, expression_string, parsed_expression_string, column_id_map]);
+            const validated = [expression_alias, expression_string, parsed_expression_string, column_id_map];
+
+            // Check if this expression is already in the array, if so then
+            // we need to replace the expression so the last expression tagged
+            // with the alias is the one that is applied to the engine.
+            if (expression_idx_map[expression_alias] !== undefined) {
```

Comment:
This is impossible without a malformed `config`, no?

Reply:
This is possible in the current UI, if one creates an expression column and then replaces it with another one - the viewer API will pass both expressions sequentially: `["// a \n 123", "// a\n upper('abcde')"]`. With all the expression UI/fix PRs merged we can fix this on the viewer side.